test/lib.sh

```diff
@@ -59,3 +59,98 @@ EOF
 hr() {
     echo "========================="
 }
+
+## diff BASELINE
+#
+# Recursively compare the current repository content to the BASELINE repository
+#
+# :param BASELINE: An existing directory used as reference to compared sync'ed files
+diff() {
+	for file in `ls` ; do 
+		if [ -d $file ] ; then 
+			pushd $file  > /dev/null
+			diff $1/$file 
+			popd  > /dev/null
+		elif [ -f $file ] ; then
+			cmp $file $1/$file
+			if [ ! $? = 0 ] ; then
+				handle_error_counter INC
+			fi
+		fi
+	done
+}
+
+## compare FOLDER
+#
+# Check FOLDER is properly sync'ed, determining the reference base on FOLDER 
+#
+# :param FOLDER: An existing directory sync'ed by boilerplate and to be checked
+compare() {
+	if [ $1 = "_data" ] ; then
+		if [ ! -f $repo/boilerplate/_data/last_boilerplate_commit ] ; then
+			# TODO: Check the content of the file to ensure it contains the proper commit in addition to the file existence
+			handle_error_counter INC
+		fi
+	else
+		pushd $1  > /dev/null
+		diff $BOILERPLATE_GIT_REPO/boilerplate/$1
+		popd > /dev/null
```

Comment:
As noted above, why can't this just be

```suggestion
		diff --recursive $BOILERPLATE_GIT_REPO/boilerplate/$1 $1 || handle_error_counter INC
```

...where by `diff` I mean the `diff` program; and we don't need the `diff` function.

Reply:
going through the `man`, looks like `-q` would provide the needed details (number of diff and which diff). Will update the script